apps/dcgm.go

```diff
@@ -0,0 +1,61 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package apps
+
+import (
+	"github.com/GoogleCloudPlatform/ops-agent/confgenerator"
+	"github.com/GoogleCloudPlatform/ops-agent/confgenerator/otel"
+)
+
+type MetricsReceiverDcgm struct {
+	confgenerator.ConfigComponent `yaml:",inline"`
+	confgenerator.MetricsReceiverShared `yaml:",inline"`
+
+	Endpoint string `yaml:"endpoint" validate:"omitempty"`
+}
+
+const defaultDcgmEndpoint = "localhost:5555"
+
+func (r MetricsReceiverDcgm) Type() string {
+	return "dcgm"
+}
+
+func (r MetricsReceiverDcgm) Pipelines() []otel.Pipeline {
+   if r.Endpoint == "" {
+      r.Endpoint = defaultDcgmEndpoint
+   }
+
+	return []otel.Pipeline{{
+		Receiver: otel.Component{
+			Type: "dcgm",
+			Config: map[string]interface{}{
+				"collection_interval": r.CollectionIntervalString(),
+            "endpoint": r.Endpoint,
+			},
```

Comment:
Try this to disable the receiver from scraping of those metrics:
```suggestion
			Config: map[string]interface{}{
				"collection_interval": r.CollectionIntervalString(),
				"endpoint":            r.Endpoint,
				"metrics": map[string]interface{}{
					"dcgm.gpu.utilization": map[string]bool{
						"enabled": false,
					},
					"dcgm.gpu.memory.bytes_used": map[string]bool{
						"enabled": false,
					},
				},
			},
```

Reply:
Is it possible to enable these in the Ops Agent configuration just for integration test? If not, I'm OK to omit it from the integration test since there are scrape tests for the receiver itself.

Reply:
We can add configuration fields to optionally enable this, if we believe there are users who would love to include those two kinds of metrics. Otherwise I don't think we can just enable those for the integration tests. 
For example: 
```
type MetricsReceiverDcgm struct {
	confgenerator.ConfigComponent       `yaml:",inline"`
	confgenerator.MetricsReceiverShared `yaml:",inline"`

	Endpoint              string `yaml:"endpoint" validate:"omitempty"`
	CollectGpuUtilization bool   `yaml:"collect_gpu_utilization"`
	CollectMemory         bool   `yaml:"collect_memory"`
}
```

Reply:
Hm. OK, in that case I'd suggest let's skip them from the integration test.